File: frontend/types/analytics.ts
export interface LeadsByStatusItem {
  status: string;
  count: number;
}

export interface LeadsByProductItem {
  product_id: string;
  product_name: string;
  count: number;
}

export interface LeadsByIndustryItem {
  industry: string;
  count: number;
}

export interface TeamPerformanceItem {
  user_id: string;
  full_name: string;
  assigned_count: number;
  won_count: number;
  lost_count: number;
  win_rate: number;
  average_score: number;
}

export interface RecentActivityItem {
  lead_id: string;
  lead_name: string;
  activity_type: string;
  description: string;
  created_at: string;
}

export interface DashboardSummary {
  scope: "organization" | "me";
  total_leads: number;
  new_leads: number;
  assigned_leads: number;
  contacted_leads: number;
  won_leads: number;
  lost_leads: number;
  high_priority_leads: number;
  average_lead_score: number;
  total_companies_researched: number;
  research_jobs_running: number;
  research_jobs_completed: number;
  leads_by_status: LeadsByStatusItem[];
  leads_by_product: LeadsByProductItem[];
  leads_by_industry: LeadsByIndustryItem[];
  team_performance: TeamPerformanceItem[];
  recent_activities: RecentActivityItem[];
}

export interface LeadsFunnelItem {
  status: string;
  count: number;
}

export interface LeadsAnalytics {
  scope: "organization" | "me";
  funnel: LeadsFunnelItem[];
  total_won: number;
  total_lost: number;
  win_rate: number;
  average_score_by_priority: Record<string, number>;
}

export interface ProductPerformanceItem {
  product_id: string;
  product_name: string;
  lead_count: number;
  average_score: number;
  won_count: number;
  win_rate: number;
}

export interface ProductsAnalytics {
  products: ProductPerformanceItem[];
}

export interface TeamAnalytics {
  team: TeamPerformanceItem[];
}

File: backend/app/schemas/analytics.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class LeadsByStatusItem(BaseModel):
    status: str
    count: int


class LeadsByProductItem(BaseModel):
    product_id: uuid.UUID
    product_name: str
    count: int


class LeadsByIndustryItem(BaseModel):
    industry: str
    count: int


class TeamPerformanceItem(BaseModel):
    user_id: uuid.UUID
    full_name: str
    assigned_count: int
    won_count: int
    lost_count: int
    win_rate: float
    average_score: float


class RecentActivityItem(BaseModel):
    lead_id: uuid.UUID
    lead_name: str
    activity_type: str
    description: str
    created_at: datetime


class DashboardSummary(BaseModel):
    """Backs both the spec §8 Organization Dashboard and Sales Representative Dashboard —
    `scope` tells the frontend which one it's looking at. "me" scope filters every lead-derived
    field to leads assigned to the caller; `team_performance` (a cross-user view) is only ever
    populated for "organization" scope, gated on the analytics.view_org permission."""

    scope: str
    total_leads: int
    new_leads: int
    assigned_leads: int
    contacted_leads: int
    won_leads: int
    lost_leads: int
    high_priority_leads: int
    average_lead_score: float
    total_companies_researched: int
    research_jobs_running: int
    research_jobs_completed: int
    leads_by_status: list[LeadsByStatusItem]
    leads_by_product: list[LeadsByProductItem]
    leads_by_industry: list[LeadsByIndustryItem]
    team_performance: list[TeamPerformanceItem]
    recent_activities: list[RecentActivityItem]


class LeadsFunnelItem(BaseModel):
    status: str
    count: int


class LeadsAnalytics(BaseModel):
    scope: str
    funnel: list[LeadsFunnelItem]
    total_won: int
    total_lost: int
    win_rate: float
    average_score_by_priority: dict[str, float]


class ProductPerformanceItem(BaseModel):
    product_id: uuid.UUID
    product_name: str
    lead_count: int
    average_score: float
    won_count: int
    win_rate: float


class ProductsAnalytics(BaseModel):
    products: list[ProductPerformanceItem]


class TeamAnalytics(BaseModel):
    team: list[TeamPerformanceItem]
